File: android/app/src/main/java/com/hsklearn/app/data/model/AiReading.kt
package com.hsklearn.app.data.model

import kotlinx.serialization.Serializable

@Serializable
data class GeneratedPassage(
    val passage: String,
    val passagePinyin: String,
    val topic: String,
)

@Serializable
data class PassageTranslation(
    val passage: String,
    val passagePinyin: String,
    val translation: String,
    val topic: String,
)

@Serializable
data class DiscussionQuestion(
    val questionChinese: String,
    val questionPinyin: String = "",
    val questionEnglish: String,
    val topic: String,
)

@Serializable
data class SpokenAnswerEvaluation(
    val overallScore: Int,
    val recognisedText: String,
    val grammar: String,
    val content: String,
    val pronunciation: String,
    val suggestions: List<String>,
    val encouragement: String,
)

@Serializable
data class TranslateRequest(
    val passage: String,
    val passagePinyin: String,
    val topic: String,
)

@Serializable
data class GenerateRequest(
    val level: Int,
)

@Serializable
data class EvaluateAnswerRequest(
    val question: String,
    val recognisedText: String,
    val level: Int,
)

File: backend/HskLearn.Api/Models/AiReading.cs
namespace HskLearn.Api.Models;

public record GeneratedPassage(
    string Passage,
    string PassagePinyin,
    string Topic
);

public record PassageTranslation(
    string Passage,
    string PassagePinyin,
    string Translation,
    string Topic
);

public record DiscussionQuestion(
    string QuestionChinese,
    string QuestionPinyin,
    string QuestionEnglish,
    string Topic
);

public record SpokenAnswerEvaluation(
    int OverallScore,
    string RecognisedText,
    string Grammar,
    string Content,
    string Pronunciation,
    List<string> Suggestions,
    string Encouragement
);
